components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -585,4 +588,28 @@ private void removeTokenFromQuery(Map<String, List<String>> parameters) {
         // remove trailing '?' or '&' from the built string
         uri = queryBuilder.substring(0, queryBuilder.length() - 1);
     }
+    private SignedJWTInfo getSignedJwtInfo(String accessToken) throws ParseException {
+
+        String signature = accessToken.split("\\.")[2];
+        SignedJWTInfo signedJWTInfo = new SignedJWTInfo();
+        Cache gatewaySignedJWTParseCache = CacheProvider.getGatewaySignedJWTParseCache();
+        if (gatewaySignedJWTParseCache != null) {
+            Object cachedEntry = gatewaySignedJWTParseCache.get(signature);
+            if (cachedEntry != null) {
+                signedJWTInfo = (SignedJWTInfo) cachedEntry;
+            } else {
+                SignedJWT signedJWT = SignedJWT.parse(accessToken);
+                signedJWTInfo.setSignedJWT(signedJWT);
+                signedJWTInfo.setJwtClaimsSet(signedJWT.getJWTClaimsSet());
+                signedJWTInfo.setToken(accessToken);
+                gatewaySignedJWTParseCache.put(signature, signedJWTInfo);
+            }
+        } else {
+            SignedJWT signedJWT = SignedJWT.parse(accessToken);
+            signedJWTInfo.setSignedJWT(signedJWT);
```

Comment:
Repeating block. Better move this to SignedJWTInfo class as mentioned in the previous comment.